### ISSUE TITLE
depthTexture packing not working: only contains data in red channel

### ISSUE THREAD

**Issue by xtassin** [AnalyticalGraphicsInc/cesium#7743]
The depthTexture, when used from a postProcessingStage shader appears to only contain data in the red channel, when depth value should be packed across all channels and unpacked with unpackDepth()

A link to a test in Sandcastle: this outputs the raw depthTexture:

https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/#c=bVPbbtswDP0VIS92sFTObS3WXLAu2Z42rECGPRkIFJt2hOliSLK7rMi/j7LcNElrGAZJkYfiOXTDDGk4PIEhC6LgiazA8lrS320sjrLWXWnlGFdgogF5ThXBx+51LfIHxSVzQO6JMzWk6tifpSpVSRIwqc1AARW6ZIa7veTZGiq3/1IXRduwYMJCW9HgPSptueNa4UF3ixUzDi2mJrQwWq6hNAA2vhmNJ3R4N53ejj4NyHRKhx+Hk7vhrW/ugWojECOiNDm9D1Vlkw2TlYA1cyyROgdhk9DnB1OdtUWTlmIXeaQwgjMs+wP5V+W4OyBsFwbvc7CU5XnccaKY9FRg+0EInCa6P5ndSdsfw12lf2rDQ3EIHU9s+pEKw0qJPTd7loPZ6NpkQBYhM6oVL7SRxLbzmfGa5J7mX/DX1QZmJE1VRD50yYh24KokDWRj0mxdSFppbXKuUEt7na95TiSqH3urf3n4fOn6D+JOQ38kq0Mfr+PzGw3e7dufvQUrxfYbTr7SQvt9aUGu0o6t+6pXWDmk2z0ajetrN46VnU5nC/54lfCi4SXRKMh7zHtpvDa9QW9u3UHAMhR/5rLSxnkRY1w6B6iHHy3Z1bhDjmbW+rJ58lI0z3lDeL5Ie1c/WtojmWDW4klRC7Hh/yDtLecJ5l+UCc2QvvJnA0awg0/Zj5bfQ5BSOk/QfVvltBY7Zs4Q/wM

**Comment by LHolst** [comment]
the demo crashed for me

![image](https://user-images.githubusercontent.com/10059243/56137129-6bce6e00-5f94-11e9-93e6-c96dfbb8e682.png)

**Comment by likangning93** [comment]
Hi @xtassin, did you mean to leave out `czm_unpackDepth` from the shader source in the Sandcastle link above? Here's what I see if I include it, this is Chrome + ANGLE on a GT 750M:

![unpackDepth](https://user-images.githubusercontent.com/5395255/56138116-327e0a00-5f64-11e9-952f-6142712d95ec.PNG)

Modified Sandcastle: https://cesiumjs.org/Cesium/Build/Apps/Sandcastle/#c=bVPJbtswEP0VwhfJqEt5a4LGC5ra7SlFA7joSYBBSyOZKBeBpJQ6gf+9Q1FxbSeCIMz6ZnmjhhnScHgCQxZEwRNZgeW1pL9bWxxlrbrSyjGuwEQD8pIqgo/d61rk94pL5oDcEWdqSNWxP0tVqpIkYFKbgQIqdMkMd3vJszVUbv+1Loq2YMGEhTajwT4qbbnjWqGj62LFjEOJqQktjJZrKA2AjT+OxhM6vJ1Ob0afB2Q6pcNPw8nt8MYX90C1EYgRUZqc3vuqssmGyUrAmjmWSJ2DsEmo84OpTtqiSEuxizxSGMEZlv2B/Jty3B0QtjOD1zlYyvI87naimPSrwPKDYDhNdHcSO09bH81dpn9qw0NyMB1P2/QjFYaVEmtu9iwHs9G1yYAsQmRUK15oI4lt5zPjNcn9mn/BX1cbmJE0VRH50AUj2oGrkjSQjUmzdSFopbXJuUIu7XW85jmRyH7spf6l8+VS9Z9CaOZCA7it7Flua1XhClvm467ceB2ftzh4t5F+f/YWvhTb77iLlRbaXxBOMQ1Q18HHVv3PYzhFpME9Go1nbTeOlR1/Z4f/eBXwyu0lAUjUe4x4yjxnvUFvbt1BwDIkf+Gy0sZ5cmM8RgfIk58w2dV4W45m1vq0efKaNM95Q3i+SHtXP2DaI5lg1qKnqIXY8GdIe8t5gvEXaUgCbrH82YAR7OBD9qPlQzBSSucJqm+znNZix8wZ4j8

**Comment by xtassin** [comment]
Yes @likangning93, it was deliberate. I tried to highlight the fact that only the red channel of the texture seems to contain data.

**Comment by likangning93** [comment]
Ah I see, that does seem a little strange. For what it's worth, we're using the technique described here for packing: http://aras-p.info/blog/2009/07/30/encoding-floats-to-rgba-the-final/

@bagnell any thoughts?

**Comment by xtassin** [comment]
I bumped into this issue again while writing a post processing fragment shader. I am doing deferred rendering and need to retrieve a world position from the depth texture. I do it a such:

`   vec4 rawDepthColor = texture2D(depthTexture, v_textureCoordinates);
    float depth = czm_unpackDepth(rawDepthColor);
    vec4 positionEC = czm_windowToEyeCoordinates(gl_FragCoord.xy, depth);
    vec4 worldCoordinate = czm_inverseView * positionEC;
    vec3 vWorldPosition = worldCoordinate.xyz / worldCoordinate.w;`

This works fine on laptop/desktop GPU but fails on mobile.

Investigation lead me to the previously reported issue. On my laptop, rawDepthColor only has meaningful data on the red channel.

On mobile, rawDepthColor has data across all four components.

Incidentally, when using only the red channel, my position retrieval code above works on both platforms.

Hope this can help.

**Comment by xtassin** [comment]
Apparently, the issue occurs (only red channel data) when GL_EXT_frag_depth and LOG_DEPTH are defined.
Also, when logarithmicDepthBuffer is disabled and multiple frustum are specified, only the first frustum seems to be used to write to the depth texture.
